docs/android/linking.md

```diff
@@ -0,0 +1,155 @@
+[DynamicDependency("MethodOnDifferentType()", typeof(ContainingType))]
+[DynamicDependency("MemberName")]
+[DynamicDependency("MemberOnUnreferencedAssembly", "ContainingType", "UnreferencedAssembly")]
+[DynamicDependency("MemberName", "Namespace.ContainingType.NestedType", "Assembly")]
+// generics
+[DynamicDependency("GenericMethodName``1")]
+[DynamicDependency("GenericMethod``2(``0,``1)")]
+[DynamicDependency("MethodWithGenericParameterTypes(System.Collections.Generic.List{System.String})")]
+[DynamicDependency("MethodOnGenericType(`0)", "GenericType`1", "UnreferencedAssembly")]
+[DynamicDependency("MethodOnGenericType(`0)", typeof(GenericType<>))]
+```
+
+## Preserve assemblies
+
+It's possible to specify assemblies that should be excluded from the linking process, while allowing other assemblies to be linked.
+
+When it links all assemblies, you can tell the linker to skip an assembly by setting the `TrimmerRootAssembly` MSBuild property in an `<ItemGroup>` tag in the project file:
+
+```xml
+<ItemGroup>
+  <TrimmerRootAssembly Include="MyAssembly" />
+</ItemGroup>
+```
+
+> [!NOTE]
+> The `.dll` extension isn't required when setting the `TrimmerRootAssembly` MSBuild property.
+
+If the linker skips an assembly, it's considered *rooted*, which means that it and all of its statically understood dependencies will be kept. You can skip additional assemblies by adding more `TrimmerRootAssembly` MSBuild properties to the `<ItemGroup>`.
+
+## Preserve assemblies, types, and members
+
+You can pass the linker an XML description file that specifies which assemblies, types, and members need to be retained.
+
+To exclude a member from the linking process when linking all assemblies, set the `TrimmerRootDescriptor` MSBuild property in an `<ItemGroup>` tag in the project file to the XML file that defines the members to exclude:
+
+```xml
+<ItemGroup>
+  <TrimmerRootDescriptor Include="MyRoots.xml" />
+</ItemGroup>
+```
+
+The XML file then uses the trimmer [descriptor format](https://github.com/dotnet/linker/blob/main/docs/data-formats.md#descriptor-format) to define which members to exclude from linking:
+
+```xml
+<linker>
+  <assembly fullname="MyAssembly">
+    <type fullname="MyAssembly.MyClass">
+      <method name="DynamicallyAccessedMethod" />
+    </type>
+  </assembly>
+</linker>
+```
+
+In this example, the XML file specifies a method that's dynamically accessed by the app, which is excluded from linking.
+
+When an assembly, type, or member is listed in the XML, the default action is preservation, which means that regardless of whether the linker thinks it's used or not, it's preserved in the output.
+
+> [!NOTE]
+> The preservation tags are ambiguously inclusive. If you don’t provide the next level of detail, it will include all the children. If an assembly is listed without any types, then all the assembly’s types and members will be preserved.
+
+## Mark an assembly as linker safe
```

Comment:
Should we have a general `[!NOTE]` section saying you should really know what you're doing if doing this? So far only Microsoft internal folks and maybe [Uno](https://platform.uno/) have done this successfully.

Library authors would probably want to enable these settings;
```xml
<SuppressTrimAnalysisWarnings>false</SuppressTrimAnalysisWarnings>
<TrimmerSingleWarn>false</TrimmerSingleWarn>
```
And then solve all the warnings that result.

I don't know if we have any existing docs on doing this. But this issue talks about it a bit:

https://github.com/xamarin/xamarin-android/issues/5652